### DISCUSSION TITLE
Is `wal_log_hints` required?

### DISCUSSION THREAD

**Discussion by ibotty** [cloudnative-pg/cloudnative-pg#4026] [General]
We have a workload that generates very big temporary tables and operates heavily on them, reading from regular logged tables.  This generates an enormous amount of WAL.

https://stackoverflow.com/questions/73445894/next-query-after-large-delete-is-slow-why-does-a-select-trigger-wal-file-archiv says, that it's that setting which triggers that behavior.

What are the downsides of potentially turning off `wal_log_hints`?  I noticed, that this setting is fixed in the operator source code (as do other operators).  So I assume it is very important.

**Comment by gbartolini** [comment]
It is required by [pg_rewind](https://www.postgresql.org/docs/current/app-pgrewind.html):

> pg_rewind requires that the target server either has the wal_log_hints option enabled in postgresql.conf or data checksums enabled when the cluster was initialized with initdb. Neither of these are currently on by default. full_page_writes must also be set to on, but is enabled by default.

And pg_rewind is required to ensure a former primary is resynced.
